website/src/pages/DailyPage/components/DailyCharts.tsx

```diff
@@ -0,0 +1,307 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { ChartContainer, ChartTooltip } from "@/components/ui/chart";
+import { Skeleton } from "@/components/ui/skeleton";
+import { MacroData, Workloads } from "@/types";
+import useApiCall from "@/utils/Hook";
+import { secondToMicrosecond } from "@/utils/Utils";
+import { useEffect, useState } from "react";
+import { CartesianGrid, Legend, Line, LineChart, XAxis, YAxis } from "recharts";
+
+export type DailyChartsProps = {
+  benchmarkType: string;
+};
+
+type NumberDataPoint = { x: string; y: number };
+type StringDataPoint = { x: string; y: string };
+
+type ChartDataItem =
+  | { id: string; data: NumberDataPoint[] }
+  | { id: string; data: StringDataPoint[] };
+
+type ChartDatas = {
+  data: ChartDataItem[];
+  title: string;
+  colors: string[];
+};
+
+const chartConfig: { [key: string]: { label: string; color: string } } = {
+  qpsReads: {
+    label: "Reads",
+    color: "hsl(var(--chart-qpsReads))",
+  },
+  qpsWrites: {
+    label: "Writes",
+    color: "hsl(var(--chart-qpsWrites))",
+  },
+  qpsOther: {
+    label: "Other",
+    color: "hsl(var(--chart-qpsOther))",
+  },
+  qpsTotal: {
+    label: "Total",
+    color: "hsl(var(--chart-qpsTotal))",
+  },
+  tps: {
+    label: "TPS",
+    color: "hsl(var(--chart-tps))",
+  },
+  latency: {
+    label: "Latency",
+    color: "hsl(var(--chart-latency))",
+  },
+  cpuTimeTotal: {
+    label: "Total",
+    color: "hsl(var(--chart-cpuTimeTotal))",
+  },
+  cpuTimeVtgate: {
+    label: "Vtgate",
+    color: "hsl(var(--chart-cpuTimeVtgate))",
+  },
+  cpuTimeVttablet: {
+    label: "Vttablet",
+    color: "hsl(var(--chart-cpuTimeVttablet))",
+  },
+  memBytesVtgate: {
+    label: "Vtgate",
+    color: "hsl(var(--chart-memBytesVtgate))",
+  },
+  memBytesVttablet: {
+    label: "Vttablet",
+    color: "hsl(var(--chart-memBytesVttablet))",
+  },
+  memBytesTotal: {
+    label: "Total",
+    color: "hsl(var(--chart-memBytesTotal))",
+  },
+};
+
+export default function DailyCharts(props: DailyChartsProps) {
+  const { benchmarkType } = props;
+  const workloads: Workloads[] = [
+    "OLTP",
+    "OLTP-READONLY",
+    "OLTP-SET",
+    "TPCC",
+    "TPCC_FK",
+    "TPCC_UNSHARDED",
+    "TPCC_FK_UNMANAGED",
+  ];
```

Comment:
I don't think we should hardcode this.

Reply:
Actually, looking at this again, we should not send all the workloads to the API. Only one is enough, since we are only displaying a single workload. Currently, we send all the workload no matter what workload was selected by the user. And we end up only looking at the data from the first element of the response (`OLTP`), even if the user selected i.e. `TPCC`. You can reproduce this by going to http://localhost/daily?type=TPCC_FK_UNMANAGED, the numbers/results will be the same as the `OLTP` workload.

Reply:
Actually I am not sure what workload's results we end up using, but this is wrong.